### PR TITLE
Improve getting recent files performance

### PR DESCRIPTION
**Pull request by CarlSchwan**
Only look for files modified in the last 2 weeks

### CODE REVIEW

File: lib/private/Files/Node/Folder.php [hunk 1]
```diff
@@ -413,37 +413,65 @@ public function getNonExistingName($name) {
...
+					'size',
+					0
 				),
 			]
 		);
+
+		$filterNonRecentFiles = new SearchComparison(
+			ISearchComparison::COMPARE_GREATER_THAN,
+			'mtime',
+			strtotime("-2 week")
```
Comment (PVince81) [inline lib/private/Files/Node/Folder.php:441]: assuming this works correctly with timezone?
or should it reset the time component to make it focus on days only ?

**Review by PVince81 — APPROVED** [review]
:+1:

File: lib/private/Files/Node/Folder.php [hunk 2]
```diff
@@ -413,37 +413,65 @@ public function getNonExistingName($name) {
...
+				$limit,
+				$offset,
+				[
+					new SearchOrder(
+						ISearchOrder::DIRECTION_DESCENDING,
+						'mtime'
+					),
+				]
+			);
+		} else {
```
[the remark this answers is not in the document]
Comment (CarlSchwan) [reply lib/private/Files/Node/Folder.php:461]: In theory yeah, that could cause some issues. If from 0 to 50 there are fewer than 40 results, then the search will skip the results from 40 to 50. 

In practice, getRecent is never called with an offset. the recent view ask for the first 100, and the recommendation apps ask for the first 7. Both use 0 as an offset in any case.

**Review by ChristophWurst — APPROVED** [review]
Reasonable trade-off

After all it says *recent* and not *most recent* :wink:

**Comment by AndyScherzinger** [comment]
@CarlSchwan so what are the changes besides only looking at files changed in the last two weeks, would 'changed' also include 'created' and is it now limited to the users storage or just the 2 weeks time frame?

Just to check with @karlitschek and @jancborchardt if it is feasible to limit recent files search in this way that it at least works again on larger instances and enables the recommendation app to work again plus in both scenarios not nuke large installations.

**Comment by karlitschek** [comment]
Sorry. I'm too stupid :-) What does this mean exactly? Is this the recent files view in the files app only? Or for other places too?

**Comment by max-nextcloud** [comment]
> @CarlSchwan so what are the changes besides only looking at files changed in the last two weeks, would 'changed' also include 'created' and is it now limited to the users storage or just the 2 weeks time frame?

The only change is the 2 weeks time frame. Group folders are included and changed should include created - haven't tested that though.

> Just to check with @karlitschek and @jancborchardt if it is feasible to limit recent files search in this way that it at least works again on larger instances and enables the recommendation app to work again plus in both scenarios not nuke large installations.

This will hopefully reduce the database load. On one instance I investigated the main load was sorting the results by timestamp to then pick the 7 most recent. The list of files to sort should be much smaller if only the ones changed in the last two weeks are included... whether it's enough to enable the app again on large instances remains to be seen.

> Sorry. I'm too stupid :-) What does this mean exactly? Is this the recent files view in the files app only? Or for other places too?

Basically all places that query for recent files. Definitely in the recommendations app.

**Comment by AndyScherzinger** [comment]
> So should I merge it? :p

Yes, fine by Frank and me behavior-wise

**Comment by karlitschek** [comment]
I think it's cool!  I like it!!

**Comment by Jerome-Herbinet** [comment]
Will this also speed up getting recent files in the Android app ?

**Comment by AndyScherzinger** [comment]
> Will this also speed up getting recent files in the Android app ?

I am not aware of the app having a recent files view.

**Comment by Jerome-Herbinet** [comment]
> > Will this also speed up getting recent files in the Android app ?
> 
> I am not aware of the app having a recent files view.

Oh, sorry, I've just seen that I've confused it with the favourites (which, in the Android application, are quite long to display). Forget about my message.
